src/edfi_amt_data_lake/parquet/amt/base/student_program_dim/main.py

```diff
@@ -0,0 +1,124 @@
+# See the LICENSE and NOTICES files in the project root for more information.
+
+from decouple import config
+
+from edfi_amt_data_lake.parquet.Common.functions import getEndpointJson
+from edfi_amt_data_lake.parquet.Common.pandasWrapper import (
+    get_descriptor_code_value_from_uri,
+    jsonNormalize,
+    pdMerge,
+    renameColumns,
+    saveParquetFile,
+    subset,
+    to_datetime_key,
+)
+
+ENDPOINT_PROGRAMS = 'programs'
+ENDPOINT_STUDENT_SCHOOL_ASSOCIATION = "studentSchoolAssociations"
+ENDPOINT_STUDENT_PROGRAM_ASSOCIATIONS = 'studentProgramAssociations'
+
+
+def student_program_dim(school_year: str) -> None:
+
+    programs_json = getEndpointJson(ENDPOINT_PROGRAMS, config('SILVER_DATA_LOCATION'), school_year)
+    student_program_dim_json = getEndpointJson(ENDPOINT_STUDENT_PROGRAM_ASSOCIATIONS, config('SILVER_DATA_LOCATION'), school_year)
+    student_school_association_json = getEndpointJson(ENDPOINT_STUDENT_SCHOOL_ASSOCIATION, config('SILVER_DATA_LOCATION'), school_year)
+
+    student_program_dim = jsonNormalize(
+        student_program_dim_json,
+        recordPath=None,
+        meta=[],
+        metaPrefix=None,
+        recordPrefix=None,
+        errors="ignore",
+    )
+
+    student_school_association = jsonNormalize(
+        student_school_association_json,
+        recordPath=None,
+        meta=[],
+        metaPrefix=None,
+        recordPrefix=None,
+        errors="ignore",
+    )
+
+    programs = jsonNormalize(
+        programs_json,
+        recordPath=None,
+        meta=["ProgramTypeDescriptor", "ProgramTypeDescriptorId", "EducationOrganizationId", "ProgramName"],
+        metaPrefix=None,
+        recordPrefix=None,
+        errors="ignore",
+    )
+
+    prev_data_frame = pdMerge(
+        left=student_program_dim,
+        right=student_school_association,
+        how='left',
+        leftOn=['studentReference.studentUniqueId'],
+        rigthOn=['studentReference.studentUniqueId'],
+        suffixLeft="_left",
+        suffixRight=None,
+    )
+
+    data_frame = pdMerge(
+        left=prev_data_frame,
+        right=programs,
+        how='left',
+        leftOn=['graduationPlanReference.educationOrganizationId'],
+        rigthOn=['educationOrganizationReference.educationOrganizationId'],
+        suffixLeft="_prev_left",
+        suffixRight=None,
+    )
+
+    get_descriptor_code_value_from_uri(data_frame, 'programReference.programTypeDescriptor')
+
+    data_frame = subset(data_frame, [
+        'beginDate',
+        'educationOrganizationReference.educationOrganizationId_prev_left',
+        'programReference.educationOrganizationId',
+        'programReference.programTypeDescriptor',
+        'studentReference.studentUniqueId',
+        'schoolReference.schoolId',
+        'graduationPlanReference.educationOrganizationId',
+    ])
+
+    data_frame['beginDate'] = to_datetime_key(data_frame, 'beginDate')
+
+    data_frame['StudentSchoolProgramKey'] = (
```

Comment:
We are missing the program name in this concatenation. That means you have to add it to the subset above as well.